doc/design/upgrade.md

```diff
@@ -1,20 +1,37 @@
 # Upgrade
 
-After installed VCH, VIC admin would want to upgrade existing instances to latest version. This document describes the upgrade requirement, and design options.
+After installed VCH, VIC admin would want to upgrade existing instances to latest version, to get new features or to fix exposed security issues. This document describes the upgrade requirement, and design options.
 
 ## Requirements
 
 ### Scope
 
-While a new version is ready, new version's vic-machine upgrade can upgrade one running or powered off VCH appliance. The containers managed by this VCH will be left in existing version. To have a new version container, user need to create new container from upgraded VCH instance. If existing container is stopped and restarted, the old version should be kept.
+While a new version is ready, new version's vic-machine can upgrade one running or powered off VCH appliance.
 
-VIC version includes three parts, "release tag"-"build id"-"github commit hash".
+For VCH appliance, vic-machine upgrade will detect the version difference between old VCH and itself, find out the metadata change in those two versions, and then migrate metadata, replace iso files. If anything wrong happens in this process, revert back to old version and status. VCH will have downtime during upgrade, but should not be broken after this process.
```

Comment:
Suggestion: 
For the VCH appliance, vic-machine upgrade will detect the version difference between the old VCH and itself. This includes detecting the guestinfo changes between the two versions. Then vic-machine upgrade will migrate the guestinfo metadata and replace the iso files. If anything incorrect happens during this process, vic-machine upgrade will revert back to old version and status. The VCH will have downtime during upgrade, but should not be broken after this process.